tf2_ros/src/tf2_ros/transform_listener.py

```diff
@@ -27,45 +27,66 @@
 
 # author: Wim Meeussen
 
-import roslib; roslib.load_manifest('tf2_ros')
-import rospy
+from rclpy.executors import SingleThreadedExecutor
+from rclpy.qos import QoSProfile
 import tf2_ros
 from tf2_msgs.msg import TFMessage
+from threading import Thread
 
-class TransformListener():
+class TransformListener:
     """
     :class:`TransformListener` is a convenient way to listen for coordinate frame transformation info.
     This class takes an object that instantiates the :class:`BufferInterface` interface, to which
     it propagates changes to the tf frame graph.
     """
-    def __init__(self, buffer):
+    def __init__(self, buffer, node, spin_thread=True, qos=QoSProfile(depth=100)):
```

Comment:
@tfoote @clalancette should this class support the use-case where `TransformListener` creates it's own node?

If so, `node` should be an optional argument, defaulting to `node=None`, and invoke a `rclpy.create_node()` to populate `self.node`